frontend/app/routes/protected/person-case/first-name.tsx

```diff
@@ -1,45 +1,97 @@
+import { firstNameSchema } from '~/.server/validation';
 import { Button } from '~/components/button';
+import { ButtonLink } from '~/components/button-link';
+import { ErrorSummary } from '~/components/error-summary';
 import { InputField } from '~/components/input-field';
 import { PageTitle } from '~/components/page-title';
+import { getFixedT } from '~/i18n-config.server';
 import { handle as parentHandle } from '~/routes/protected/layout';
+import { getLanguage } from '~/utils/i18n-utils';
 
 export const handle = {
   i18nNamespace: [...parentHandle.i18nNamespace, 'protected'],
 } as const satisfies RouteHandle;
 
-export default function FirstName() {
+export async function loader({ context, request }: Route.LoaderArgs) {
+  requireAuth(context.session, new URL(request.url), ['user']);
+  const t = await getFixedT(request, handle.i18nNamespace);
+
+  return {
+    documentTitle: t('protected:person-case.page-title'),
+    sessionData: context.session.inPersonSINCase,
+  };
+}
+
+export function meta({ data }: { data: { documentTitle: string } }) {
+  return [{ title: data.documentTitle }];
+}
+
+export async function action({ context, request }: ActionFunctionArgs) {
+  requireAuth(context.session, new URL(request.url), ['user']);
+  const lang = getLanguage(request);
+
+  const formData = await request.formData();
+  const input = { firstName: formData.get('first-name') };
+
+  // Validation schema
+  const schema = v.object({ firstName: firstNameSchema() });
+
+  // Safe-parse form data input
+  const parsedDataResult = v.safeParse(schema, input, { lang });
+
+  if (!parsedDataResult.success) {
+    return data({ errors: v.flatten<typeof schema>(parsedDataResult.issues).nested }, { status: 400 });
+  }
+  // If the first name is valid, store it in the session and redirect to the next page
+
+  context.session.inPersonSINCase = input;
+  return i18nRedirect('routes/protected/person-case/last-name.tsx', request);
+}
+
+export default function FirstName({ loaderData, actionData, params }: Route.ComponentProps) {
   const { t } = useTranslation(handle.i18nNamespace);
 
+  const fetcher = useFetcher<Info['actionData']>();
+  const isSubmitting = fetcher.state !== 'idle';
+  const errors = fetcher.data?.errors;
+
   return (
     <>
       <div className="mb-8">
         <PageTitle className="mt-8">{t('protected:person-case.page-title')}</PageTitle>
-        <Form id="first-name-form" method="post">
-          <InputField
-            className="mb-4"
-            id="first-name-id"
-            label={t('protected:person-case.first-name')}
-            name="first-name"
-            required
-            type="text"
-          />
-          <div className="space-x-2">
-            <Button>{t('protected:person-case.previous')}</Button>
-            <Button>{t('protected:person-case.next')}</Button>
+        <ErrorSummary errors={errors} />
+        <fetcher.Form method="post" noValidate>
+          <div className="mb-6 space-y-4">
```

Comment:
This `div` in unnecessary because in has only one input control. Using `space-y-4` has no effect to space his children.

[Adding space between children](https://tailwindcss.com/docs/margin#adding-space-between-children)